### DISCUSSION TITLE
Organize the menu

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [mdgrs-mei/PowerShellRun#50] [General]
Is there any way to organize the menu items? For example, I'd like to have my favorite files grouped together under a parent "folder."  The same would be true of the different categories This would be especially useful if enabling all categories.

**Comment by mdgrs-mei** [comment]
Is the parent folder where you would like to have your files grouped different from the actual parent folder in the filesystem? If it's fine with the actual parent folder, you can use Favorite Folder instead.

If you want to make a grouping not related to the filesystem hierarchy, I'm thinking of this kind of API:

```powershell
$group = Add-PSRunEntryGroup -Name 'ProjectA' -Icon '💎' -PassThru
Add-PSRunFavoriteFile -Path D:\test.ps1 -EntryGroup $group
```

For the category grouping, maybe this is fine?

```powershell
Add-PSRunEntryGroup -Name 'Executables' -Icon '🔧' -Category 'Executable'
```

**Comment by jdhitsolutions** [comment]
I'm thinking "entry group", something that I could collapse or open to show items. For example, I might want to keep all of my custom functions in a group. Or favorite files. This is not file system related, but is using the file system as a model.

**Reply by mdgrs-mei** [reply]
ok, "entry group" which can have any entries as children. The UI will work just like the Favorite Folder where you can open it with `Enter` and go back with `Backspace`. The functions will be something like this https://github.com/mdgrs-mei/PowerShellRun/discussions/50#discussioncomment-10043750.

Entry groups will add another step to run the child entries compared to the default flat structure but it should be more intuitive sometimes. Personally I would add a group for the `Executable` category.

I will work on this as the next main feature release.

**Reply by mdgrs-mei** [reply]
@jdhitsolutions The EntryGroup feature was added by #55 and has been included in v0.8.0. The documentation is [here](https://github.com/mdgrs-mei/PowerShellRun?tab=readme-ov-file#-entrygroup).
